### PR TITLE
closes #2520: upgrade to quarkus 3.x

### LINKED ISSUE #2520: Update to Quarkus 3.x line
With 3.0.0.CR1 out we should start working on the update to get an idea how hard it s going to be.

### PR DESCRIPTION
**Pull request by ivansenic**
*Work in progress..*

**What this PR does**:
Opening the PR in draft that should provide us with feedback on upgrading to Quarkus 3.x line. This is work that can not be done immediately, so we want to proceed step by step.

Guides:
* https://quarkus.io/blog/quarkus-3-0-0-cr2-released/#upgrading
* https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0

**Which issue(s) this PR fixes**:
Fixes #2520 

**Checklist**
- [x] Major release version, not release candidate
- [x] Align gRPC versions coordinator & apis
- [x] Multi-part breaking changes
- [ ] Management endpoint breaking changes (optional, can be moved to separate port)
- [x] Open-telemetry config breaking changes
- [x] Fix compile issues
- [x] Unit tests green
- [x] Integration tests green
- [x] Docker image upgrades
- [x] Silent `LoggingProviderImpl`
- [x] Fix DevUI link, move start page to v3
- [x] Remove basic auth from swagger
- [x] jackson/snakeyaml version resolution (cc. @tatu-at-datastax)
- [x] need https://github.com/quarkusio/quarkus/issues/32944 released
- [ ] time zone test issue

### CODE REVIEW

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/schema/SchemaManager.java [hunk 1]
```diff
@@ -632,7 +632,14 @@ private Uni<Schema.CqlKeyspaceDescribe> getKeyspaceInternal(
                 // if it's not null, we map to keyspace with cached flag true
                 return Uni.createFrom()
                     .future(keyspaceFuture)
-                    .map(
+
+                    // if failure is cached, invalidate
+                    .onFailure()
```
Comment (jeffreyscarpenter) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/schema/SchemaManager.java:637]: Does this change some functionality?

**Review by jeffreyscarpenter — APPROVED** [review]
Looks good to me. Assume there will be a follow-on ticket to do the same for JSON API?

**Review by tatu-at-datastax — APPROVED** [review]
Vast majority of changes are for JAXB -> Jakarta API changes (imports) which are of course necessary and fine. Also a few minor version upgrades which should be fine as well (I think our test suites would catch issues with Duzzt f.ex).
And also some config name changes (opentelemetry -> otel).

Did not see anything unexpected although not very familiar with new/extended tests for consistency. Assuming they are fine (and will try to re-read just to learn).

So looks good, relatively small actual changes it sounds like.

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/schema/SchemaManager.java (hunk 1 above)
Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/schema/SchemaManager.java:637]: No, tests remained the same.. They changed how failures are cached and this we need to invalidate if we get a failure emitted from the cache.. everything should be the same as before..

**Comment by ivansenic** [comment]
> Vast majority of changes are for JAXB -> Jakarta API changes (imports) which are of course necessary and fine. Also a few minor version upgrades which should be fine as well (I think our test suites would catch issues with Duzzt f.ex). And also some config name changes (opentelemetry -> otel).
> 
> Did not see anything unexpected although not very familiar with new/extended tests for consistency. Assuming they are fine (and will try to re-read just to learn).
> 
> So looks good, relatively small actual changes it sounds like.

@tatu-at-datastax 

* new tests for consistency are not new, there were just inner classes `QueryExecutorTest$WithDifferentConsistency`, but this does not work anymore, so I just pulled them out as separate class
* `opentelemetry -> otel` is the braking change, you can check in the release notes

**Comment by ivansenic** [comment]
@tatu-at-datastax @jeffreyscarpenter can you try running this locally, check if you see anything suspicious, try swagger, try starting docker images, a bit more checks, rather than simply looking into the code?

**Comment by jeffreyscarpenter** [comment]
I built everything locally and ran all integration tests for API services (errors noted above), then built docker images and started docker compose for Cassandra 4.0. I ran Postman collections against each API with no errors. I also opened all swagger/health/metrics endpoints as listed at https://github.com/stargate/stargate/tree/main/docker-compose#exposed-endpoints.

**Comment by ivansenic** [comment]
> I got three test failures on GraphQL integration tests when running on my desktop:
> 
> ```
> [ERROR]   ScalarsIntegrationTest.shouldSupportScalar:97
> expected: "2023-05-10T21:07:35.949Z"
>  but was: "2023-05-10T14:07:35.949-07:00"
> [ERROR]   DataTypesIntegrationTest.insertAndRetrieveValue:95
> expected: "2011-02-03T04:05:00.000Z"
>  but was: "2011-02-02T21:05:00.000-07:00"
> [ERROR]   DataTypesIntegrationTest.insertAndRetrieveValue:95
> expected: "2011-02-03T04:05:00.000Z"
>  but was: "2011-02-02T21:05:00.000-07:00"
> [INFO]
> [ERROR] Tests run: 250, Failures: 3, Errors: 0, Skipped: 1
> ```

Will try to understand this and fix.. Not sure if this happens only on `3.x` or not, but your time zone does have the influence..

**Comment by ivansenic** [comment]
Rebased, updated to `3.0.3.Final` and bump maven plugin versions..

**Comment by jeffreyscarpenter** [comment]
Looks good to me, I think we should move forward.

**Pull request merged**